tests/js/spec/views/settings/components/dataScrubbing/dataScrubbing.spec.tsx

```diff
@@ -0,0 +1,364 @@
+import React from 'react';
+import sortBy from 'lodash/sortBy';
+
+import {mountWithTheme} from 'sentry-test/enzyme';
+
+import DataScrubbing from 'app/views/settings/components/dataScrubbing';
+import {
+  ProjectId,
+  MethodType,
+  RuleType,
+} from 'app/views/settings/components/dataScrubbing/types';
+import {
+  getMethodLabel,
+  getRuleLabel,
+} from 'app/views/settings/components/dataScrubbing/utils';
+import {addSuccessMessage} from 'app/actionCreators/indicator';
+
+import {convertedRules, stringRelayPiiConfig, relayPiiConfig} from './utils';
+
+jest.mock('app/actionCreators/indicator');
+
+const organizationSlug = 'sentry';
+const handleUpdateOrganization = jest.fn();
+const additionalContext = 'These rules can be configured for each project.';
+
+function getOrganization(piiConfig?: string) {
+  // @ts-ignore
+  return TestStubs.Organization(
+    piiConfig ? {id: '123', relayPiiConfig: piiConfig} : {id: '123'}
+  );
+}
+
+function renderComponent({
+  disabled,
+  projectId,
+  endpoint,
+  ...props
+}: Partial<Omit<DataScrubbing<ProjectId>['props'], 'endpoint'>> &
+  Pick<DataScrubbing<ProjectId>['props'], 'endpoint'>) {
+  const organization = props.organization ?? getOrganization();
+  if (projectId) {
+    return mountWithTheme(
+      <DataScrubbing
+        additionalContext={additionalContext}
+        endpoint={endpoint}
+        projectId={projectId}
+        relayPiiConfig={stringRelayPiiConfig}
+        disabled={disabled}
+        organization={organization}
+        onSubmitSuccess={handleUpdateOrganization}
+      />
+    );
+  }
+  return mountWithTheme(
+    <DataScrubbing
+      additionalContext={additionalContext}
+      endpoint={endpoint}
+      relayPiiConfig={stringRelayPiiConfig}
+      disabled={disabled}
+      organization={organization}
+      onSubmitSuccess={handleUpdateOrganization}
+    />
+  );
+}
+
+describe('Data Scrubbing', () => {
+  describe('Organization level', () => {
+    const endpoint = `organization/${organizationSlug}/`;
+
+    it('default render', () => {
+      const wrapper = renderComponent({disabled: false, endpoint});
+
+      //State
+      expect(wrapper.state().rules).toEqual(convertedRules);
```

Comment:
Coupling to a components state has not worked out well for me in the past. Does this state get translated into components/text/controls that are more user facing and likely more stable?

Reply:
You are right! I have just updated the tests that were using the state, with the exception of sourceField. The sourceField.spec renders the input components differently according to the newly passed props and as I am only testing the source input component this is not possible... I will write more tests afterward to cover this behavior 😉